configure.ac

```diff
@@ -403,12 +404,19 @@ AS_CASE([$host],
     mklib="rm -f \$(1) && ${AR} rc \$(1) \$(2) && ${RANLIB} \$(1)"
   ])
 
-# C preprocessor: we do not rely on AC_PROG_CPP because it returns
-# 'fcc -E' while we really need cpp
-AC_CHECK_TOOL([DIRECT_CPP],[cpp])
+# Determine how to call the C preprocessor directly.
+# Most of the time, calling the C preprocessor through the C compiler is
+# desirable and even important.
+# In some cases, though, we wnat to use the C preprocessor only to
+# extend macros. In such cases, it is much more convenient to be able
+# to invoke it directly rather than through the C compiler, for instance
+# because, when invoked directly, the C preprocessor does not require
+# to be invoked on a file with a '.c' extension
+# We thus figure out how to invoke the C preprocessor directly but
+# let the CPP variable untouched
 AS_CASE([$host],
-  [*-pc-windows], [CPP="$CC -nologo -EP"],
-  [CPP=$DIRECT_CPP])
+  [*-pc-windows], [DIRECT_CPP="$CC -nologo -EP"],
```

Comment:
Following on from our private discussion, autoconf will infer `CPP=cl -E` which is different (`-EP` suppresses `#line` directives and `-nologo` suppresses the Microsoft copyright notice). So I think the MSVC part needs to stay as it is but add `DIRECT_CPP="$CPP"`